gdpr/loading.py

```diff
@@ -1,57 +1,162 @@
 from django.conf import settings
+from django.core.exceptions import ImproperlyConfigured
+from django.db.models import Model
 from django.utils.encoding import force_text
 
-from is_core.utils import str_to_class
+from .utils import str_to_class
 
+if TYPE_CHECKING:
+    from gdpr.anonymizers import ModelAnonymizer
+    from gdpr.purposes import AbstractPurpose
 
-class AppAnonymizerLoader:
-    """
-    Anonymizers loader that interates over all applications and is looks for anonymizers.py file that should contain
-    anonymizer classes.
-    """
 
-    def import_anonymizers(self):
+class BaseLoader:
+    """Base class for all loaders."""
+
+    def import_modules(self) -> None:
+        raise NotImplementedError()
+
+
+class AppLoader(BaseLoader):
+    """Scan all installed apps for `module_name` module."""
+    module_name: str
+
+    def import_modules(self) -> None:
         for app in apps.get_app_configs():
+            if app.name == "gdpr":
+                continue
             try:
-                import_module('{}.anonymizers'.format(app.name))
+                import_module(f'{app.name}.{self.module_name}')
             except ImportError as ex:
-                if ((six.PY2 and force_text(ex) != 'No module named anonymizers') or
-                        (six.PY3 and force_text(ex) != 'No module named \'{}.anonymizers\''.format(app.name))):
+                if force_text(ex) != f'No module named \'{app.name}.{self.module_name}\'':
                     raise ex
 
 
-class AnonymizersRegister:
-    """
-    Register is storage for found anonymizer classes.
-    """
+class SettingsListLoader(BaseLoader):
+    """Import all modules from list `list_name` in settings."""
+    list_name: str
+
+    def import_modules(self):
+        if not hasattr(settings, self.list_name):
+            raise ImproperlyConfigured(f"settings.{self.list_name} not found.")
+        modules_list = getattr(settings, self.list_name)
+        if type(modules_list) is str:
+            import_module(modules_list)
+        elif type(modules_list) in [list, tuple]:
+            for i in modules_list:
+                import_module(i)
+        else:
+            raise ImproperlyConfigured(f"settings.{self.list_name} have incorrect type {str(type(modules_list))}.")
+
+
+class SettingsListAnonymizerLoader(SettingsListLoader):
+    """Load all anonymizers from settings.GDPR_ANONYMIZERS_LIST list."""
+    list_name = "GDPR_ANONYMIZERS_LIST"
+
+
+class SettingsListPurposesLoader(SettingsListLoader):
+    """Load all purposes from settings.GDPR_PURPOSES_LIST list."""
+    list_name = "GDPR_PURPOSES_LIST"
+
+
+class AppAnonymizerLoader(AppLoader):
+    """Scan all installed apps for anonymizers module which should contain anonymizers."""
+
+    module_name = "anonymizers"
+
+
+class AppPurposesLoader(AppLoader):
+    """Scan all installed apps for purposes module which should contain purposes."""
+    module_name = "purposes"
+
+
+K = TypeVar('K')
+V = TypeVar('V')
```

Comment:
tohle mi strašně na tom statickém typování vadí :( snaží se simulovat ostatní jazyky i když to moc do pythonu nepatří